### PR TITLE
Null tenant list issues

### PR DESCRIPTION
**Pull request by adamoctoclose**
If tenanted only deployments is enabled and no tenants are passed in I've added a check and fail if this is the case. I've also added a check to see if the tenant list is null before it checks to find match tenants. This is to fix an issue whereby if no tenants are in the space then the script would bring all tenants back and select the first tenant in the list and associate that deployment target with the tenant.

### CODE REVIEW

**Comment by adamoctoclose** [comment]
Essentially the script only ever worked if the space had tenants.

File: step-templates/octopus-register-listening-target.json [hunk 1]
```diff
@@ -1,135 +1,135 @@
...
+  },
+  "Parameters": [
+    {
+      "Id": "f4475bac-5fdb-4d3b-a5de-20dafb8d847e",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:17]: This looks like a new ID, did you paste in the entire JSON body or just the script?

File: step-templates/octopus-register-listening-target.json [hunk 2]
```diff
@@ -1,135 +1,135 @@
...
+      "Label": "Octopus Base Url",
+      "HelpText": "The base url of your Octopus Deploy instance.  Example: https://samples.octopus.app",
+      "DefaultValue": "",
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
+      }
+    },
+    {
+      "Id": "d25ad4b7-9281-4ee0-833d-b07f0e4b5b19",
+      "Name": "RegisterListeningTarget.Octopus.Api.Key",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:28]: This looks like a new ID, did you paste in the entire JSON body or just the script?

File: step-templates/octopus-register-listening-target.json [hunk 3]
```diff
@@ -1,135 +1,135 @@
...
+      "Label": "Octopus Api Key",
+      "HelpText": "The API key of a user in Octopus Deploy who has permissions to register the cluster.",
+      "DefaultValue": "",
+      "DisplaySettings": {
+        "Octopus.ControlType": "Sensitive"
+      }
+    },
+    {
+      "Id": "264cc481-363d-4f76-9f41-5356cf65c28c",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:37]: This looks like a new ID, did you paste in the entire JSON body or just the script?

File: step-templates/octopus-register-listening-target.json [hunk 4]
```diff
@@ -1,135 +1,135 @@
...
+      "Name": "RegisterListeningTarget.Machine.Name",
+      "Label": "Machine Name",
+      "HelpText": "The name of the machine to register with Octopus Deploy.",
+      "DefaultValue": "",
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
+      }
+    },
+    {
+      "Id": "347ee05a-7e17-4edf-a884-3dc0fa8d862b",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:47]: This looks like a new ID, did you paste in the entire JSON body or just the script?

File: step-templates/octopus-register-listening-target.json [hunk 5]
```diff
@@ -1,135 +1,135 @@
...
+      "Name": "RegisterListeningTarget.Machine.Address",
+      "Label": "Machine Address",
+      "HelpText": "The machine address (IP Address or Domain Name) to connect to",
+      "DefaultValue": "",
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
+      }
+    },
+    {
+      "Id": "bae91f6b-557a-4e59-a666-3783e8557b8c",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:57]: This looks like a new ID, did you paste in the entire JSON body or just the script?

File: step-templates/octopus-register-listening-target.json [hunk 6]
```diff
@@ -1,135 +1,135 @@
...
-        "Id": "5b2e0993-7f96-4447-b47b-029c8f225688",
-        "Name": "RegisterListeningTarget.Overwrite.Existing",
-        "Label": "Overwrite Existing Registration",
-        "HelpText": "Indicates if the existing listening tentacle should be overwritten in Octopus.",
-        "DefaultValue": "False",
-        "DisplaySettings": {
-          "Octopus.ControlType": "Checkbox"
-        }
+    {
+      "Id": "25dd9d35-448e-4536-b622-d8d2fce8d4f4",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:67]: This looks like a new ID, did you paste in the entire JSON body or just the script?

File: step-templates/octopus-register-listening-target.json [hunk 7]
```diff
@@ -1,135 +1,135 @@
...
-    "$Meta": {
-      "ExportedAt": "2020-04-13T15:37:29.866Z",
-      "OctopusVersion": "2020.1.10",
-      "Type": "ActionTemplate"
     },
-    "Category": "octopus"
-  }
\ No newline at end of file
+    {
+      "Id": "62c64e03-7fec-4cdb-8a9c-4038f767ffd2",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:77]: This looks like a new ID, did you paste in the entire JSON body or just the script?

File: step-templates/octopus-register-listening-target.json [hunk 8]
```diff
@@ -1,135 +1,135 @@
...
+      "Name": "RegisterListeningTarget.Environment.List",
+      "Label": "Environment CSV List",
+      "HelpText": "Comma separated list of environments to assign to the the machine in Octopus Deploy.",
+      "DefaultValue": "",
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
+      }
+    },
+    {
+      "Id": "8d80c23a-7f5d-43c4-885b-c6696b947d20",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:87]: This looks like a new ID, did you paste in the entire JSON body or just the script?

File: step-templates/octopus-register-listening-target.json [hunk 9]
```diff
@@ -1,135 +1,135 @@
...
+      "Name": "RegisterListeningTarget.Tenant.List",
+      "Label": "Tenant CSV List",
+      "HelpText": "(Optional) If this is for a tenant, the a comma separated list of tenants to assign the machine to in Octopus Deploy",
+      "DefaultValue": "",
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
+      }
+    },
+    {
+      "Id": "b9a9e212-b1e2-4222-b475-b6a2b170ea2b",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:97]: This looks like a new ID, did you paste in the entire JSON body or just the script?

File: step-templates/octopus-register-listening-target.json [hunk 10]
```diff
@@ -1,135 +1,135 @@
...
+      "Label": "Tenanted Deployments",
+      "HelpText": "Choose the kind of deployment where this deployment target should be included.",
+      "DefaultValue": "Untenanted",
+      "DisplaySettings": {
+        "Octopus.ControlType": "Select",
+        "Octopus.SelectOptions": "Untenanted|Exclude from tenanted deployments (default)\nTenanted|Include only in tenanted deployments\nTenantedOrUntenanted|Include in both tenanted and untenanted deployments"
+      }
+    },
+    {
+      "Id": "b2cd605b-aa01-46a0-8bff-4e0140c4aba1",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:108]: This looks like a new ID, did you paste in the entire JSON body or just the script?

File: step-templates/octopus-register-listening-target.json [hunk 11]
```diff
@@ -1,135 +1,135 @@
...
+      "Name": "RegisterListeningTarget.MachinePolicy.IdOrName",
+      "Label": "Machine Policy Id Or Name",
+      "HelpText": "Enter in the name or the Id of the Machine Policy in Octopus Deploy for the AKS Cluster.",
+      "DefaultValue": "Default Machine Policy",
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
+      }
+    },
+    {
+      "Id": "fca61fec-e270-4ba0-9c6c-af7ff28c1ffc",
```
Comment (OctopusBob) [inline step-templates/octopus-register-listening-target.json:118]: This looks like a new ID, did you paste in the entire JSON body or just the script?

**Review by OctopusBob — COMMENTED** [review]
It looks like the entire JSON body from the export was pasted in rather than just the script.  This is causing all new Ids which will break a lot of people.  Please just paste in the properties object rather than the entire JSON body.

**Review by OctopusBob — APPROVED** [review]
download

**Pull request merged**